fillers/withdrawals/withdrawals.py

```diff
@@ -0,0 +1,653 @@
+        to_address(0x8),
+        to_address(0x9),
+        to_address(2**160 - 1),
+    ]
+
+    pre = {
+        TestAddress: Account(balance=1000000000000000000000, nonce=0),
+    }
+
+    for addr in ADDRESSES:
+        pre[addr] = Account(
+            code=SET_STORAGE,
+        )
+
+    # Many repeating withdrawals of the same accounts in the same block.
+    blocks = [
+        Block(
+            withdrawals=[
+                Withdrawal(
+                    index=i,
+                    validator=0,
+                    address=ADDRESSES[i % len(ADDRESSES)],
+                    amount=10**9,
+                )
+                for i in range(len(ADDRESSES) * 16)
+            ],
+        ),
+    ]
+
+    post = {}
+
+    for addr in ADDRESSES:
+        post[addr] = Account(
+            balance=16 * 10**9,
+            storage={},
+        )
+
+    yield BlockchainTest(pre=pre, post=post, blocks=blocks)
+
+    # Similar test but now use multiple blocks each with multiple withdrawals
+    # to the same withdrawal address.
+    # Expected post is exactly the same.
+    blocks = [
+        Block(
+            withdrawals=[
+                Withdrawal(
+                    index=i * 16 + j,
+                    validator=i,
+                    address=ADDRESSES[i],
+                    amount=10**9,
+                )
+                for j in range(16)
+            ],
+        )
+        for i in range(len(ADDRESSES))
+    ]
+
+    yield BlockchainTest(pre=pre, post=post, blocks=blocks)
+
+
+@test_from(WITHDRAWALS_FORK)
+def test_withdrawals_many_withdrawals(_):
+    """
+    Test Withdrawals with a count of N withdrawals in a single block where
+    N is a high number not expected to be seen in mainnet.
+    """
+    N = 400
+    pre = {
+        TestAddress: Account(balance=1000000000000000000000, nonce=0),
+    }
+    withdrawals = []
+    post = {}
+    for i in range(N):
+        addr = to_address(0x100 * i)
+        amount = i * 10**9
+        pre[addr] = Account(
+            code=SET_STORAGE,
+        )
+        withdrawals.append(
+            Withdrawal(
+                index=i,
+                validator=i,
+                address=addr,
+                amount=amount,
+            )
+        )
+        post[addr] = Account(
+            code=SET_STORAGE,
+            balance=amount,
+            storage={},
+        )
+
+    blocks = [
+        Block(
+            withdrawals=withdrawals,
+        ),
+    ]
+
+    yield BlockchainTest(pre=pre, post=post, blocks=blocks)
+
+
+@test_from(WITHDRAWALS_FORK)
+def test_withdrawals_self_destructing_account(_):
+    """
+    Test Withdrawals can be done to the same address multiple times in
+    the same block.
+    """
+    pre = {
+        TestAddress: Account(balance=1000000000000000000000, nonce=0),
+        to_address(0x100): Account(
+            code=SELFDESTRUCT,
+            balance=100,
+        ),
+        to_address(0x200): Account(
+            balance=0,
```

Comment:
I think this is exactly what's happening:
- `0x100` has which are sent during self-destruct to `0x200`.
- Then `0x100` receives a withdrawal with 99 wei.
- Lastly the verification is made at `0x200` that the 100 wei were received from the self-destruct, and `0x100` received the withdrawal amount.

I updated the comment because it was out of date.